tinyagentos/containers/native.py

```
@@ -0,0 +1,670 @@
+# Callers MUST treat name as an opaque identifier, not a security boundary.
+# ---------------------------------------------------------------------------
+
+# Directories that push_file is allowed to write into on the host.
+# Any remote_path that resolves outside these roots is rejected.
+# This set MUST be kept minimal — add entries only when a concrete
+# deployment path genuinely needs them.
+_ALLOWED_WRITE_ROOTS: tuple[str, ...] = (
+    "/opt/taos/",
+    "/etc/systemd/system/",
+    "/var/lib/taos/",
+    "/tmp/",
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp/"

(S108)